Add colorbars for plot2D

thomasdorch opened the pull request:
This PR:
- Adds colorbars for `plot2D` (both in `plot_epsilon` and `plot_fields`) (#2286)
- Adds additional field components (and derived components) to `plot_fields` (#2267)

I think there's more room to improve the look of the colorbars, so let me know if I should make any adjustments!

So far fields look like this:
<img width="300" alt="image" src="https://user-images.githubusercontent.com/53101766/198534668-61653ac3-b2ce-492e-8938-d1b13b7762d3.png">
and epsilon looks like this:
<img width="300" alt="image" src="https://user-images.githubusercontent.com/53101766/198534880-df9c4a24-5e91-40f7-a91b-c759d8272d46.png">


@smartalecH 
@joamatab 
@SkandanC

oskooi commented on python/visualization.py:
```diff
@@ -447,6 +450,31 @@ def sort_points(xy):
+    )
+    plt.colorbar(
+        mappable=sm,
+        cax=make_axes_locatable(ax).append_axes(position="right", size="5%", pad=0.05),
```
Any reason why the `pad` parameter is chosen as an absolute quantity rather than a percentage (e.g., `2%`) similar to `size`?

thomasdorch replied:
Good catch, I agree that using the relative quantity is better here.

oskooi reviewed (approved):
LGTM.

oskooi commented:
You should also add this new feature to the `plot2D` docstrings in [`simulation.py`](https://github.com/NanoComp/meep/blob/5fa12b499d4934ea8b352e3e8033762b34f6eb0c/python/simulation.py#L4657-L4673) so that it is documented in the user manual.

smartalecH commented:
Why not pass the default plotting options used in `plot2D`?

thomasdorch commented:
Thanks for the feedback! 

I think it's in much better shape now that I've:
- updated the docs in `simulation.py` @oskooi 
- added new argument, `colorbar_parameters`, that let the user specify parameters for the colorbar (including `label`) and pass them into `plot2d` @smartalecH 

Let me know what you think.

smartalecH commented:
LGTM!
